src/extensions/UserDefinedFormControllerExtension.php

```diff
@@ -285,4 +288,138 @@ public function getShareLink()
+        $submittedData = SubmittedFormField::get()
+            ->where(['ParentID' => $submissionID]);
+
+        $stepsList = new ArrayList();
+        $fieldsList = new ArrayList();
+        $groupsList = new ArrayList();
+
+        $_fields = 0;
+        $groupTitle = '';
+
+        foreach ($formFields as $formField) {
+            if ($formField->ClassName == "SilverStripe\UserForms\Model\EditableFormField\EditableFormStep") {
+                // Store title so when new step is encountered this title will be saved
+                // For first step do not add data just continue store title in currentTitle
+                if ($_fields == 0) {
+                    $stepTitle = $formField->Title ?? 'Section';
+                    $_fields++;
+                    continue;
+                }
+
+                if ($fieldsList->count() != 0) {
+                    $groupsList->add([
+                        'FieldGroupTitle' => '',
+                        'FieldsList' => $fieldsList
+                    ]);
+                    $fieldsList = new ArrayList();
+                }
+
+                $stepsList->add([
+                    'StepTitle' => $stepTitle,
+                    'FieldGroup' => $groupsList,
+                ]);
+                $stepTitle = $formField->Title ?? 'Section';
+                $groupsList = new ArrayList();
+                continue;
+            }
+
+            if ($formField->ClassName == "SilverStripe\UserForms\Model\EditableFormField\EditableFieldGroup") {
+                // New group start detected - Store values of previous fields as ungrouped data
+                if ($fieldsList->Count()) {
+                    $groupsList->add([
+                        'FieldGroupTitle' => '',
+                        'FieldsList' => $fieldsList
+                    ]);
+                }
+                $groupTitle = $formField->Title ?? "";
+                $fieldsList = new ArrayList();
+                continue;
+            }
+            if ($formField->ClassName == "SilverStripe\UserForms\Model\EditableFormField\EditableFieldGroupEnd") {
+                $groupsList->add([
+                    'FieldGroupTitle' => $groupTitle,
+                    'FieldsList' => $fieldsList
+                ]);
+                $fieldsList = new ArrayList();
+                continue;
+            }
+
+            // Process the current field
+            $submission = $submittedData->where(['Name' => $formField->Name])->first();
+
+            // Check if file is uploaded
+            if ($formField->ClassName == "SilverStripe\UserForms\Model\EditableFormField\EditableFileField") {
+                $fieldsList->add([
+                    'FieldTitle' => $formField->Title ?? "Field Generic",
+                    // 'FieldValue' => $submission->getFileName(),  <--- Show filename - TBD
+                    'FieldValue' =>  $submission->getFileName() ? 'File uploaded' : '',
```

Comment:
```suggestion
                    'FieldValue' =>  $submission->getFileName() ? $submission->getFormattedValue() : '',
```

Reply:
I don't want to show the filename here. It will be decided later.

Reply:
Fair enough and ask the client